server/segment_rpc.go

```diff
@@ -439,7 +434,11 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 
 	glog.Infof("Successfully transcoded segment nonce=%d manifestID=%s segName=%s seqNo=%d", nonce, string(sess.ManifestID), seg.Name, seg.SeqNo)
 
-	return tdata, nil
+	return &ReceivedTranscodeResult{
+		TranscodeData: tdata,
+		Info:          tr.Info,
+		LatencyScore:  transcodeDur.Seconds() / seg.Duration,
+	}, nil
 }
 
 func updateOrchestratorInfo(sess *BroadcastSession, oInfo *net.OrchestratorInfo) {
```

Comment:
Just noting that this is now only used in `broadcast.go` , but it doesn't really matter because it still compiles.

However, if you were to move this over to broadcast.go, one change I would make is to set the signature like so:

`func updateSession(oldSess *BroadcastSession, res *ReceivedTranscodeResult) (newSess *BroadcastSession)`

And that function can create a new session, copy old data over, update the latency score, update the O info if necessary, and return the new broadcast session. That should simplify some of the boilerplate around the `updateOrchestratorInfo` call site to something like: `cxn.sessManager.completeSession(updateSession(sess, res))`

Reply:
Replaced `updateOrchestatorInfo()` with `updateSession()` (in `broadcast.go`) in 434f82f